sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/externals/ExternalHierarchyLogicImpl.java

```diff
@@ -342,6 +358,14 @@ public EvalHierarchyNode getRootLevelNode() {
      * @see org.sakaiproject.evaluation.providers.EvalHierarchyProvider#getNodeById(java.lang.String)
      */
     public EvalHierarchyNode getNodeById(String nodeId) {
+        if(cache.containsKey(nodeId)){
```

Comment:
I believe the preferred method is to do one cache.get(nodeId) and then check to see if that result is null. I believe containsKey should be avoided.